### DISCUSSION TITLE
How to install mailpit on Windows and integrate with Laragon

### DISCUSSION THREAD

**Discussion by Pen-y-Fan** [leokhoa/laragon#419] [Show and tell]
Laravel 10 has changed the default smtp client from **MailHog** to **Mailpit**. I've taken a look and documented how to install it in Laragon:

- [How to install mailpit on Windows and integrate with Laragon](https://pen-y-fan.github.io/2023/02/23/how-to-install-mailpit-in-laragon/)

**Comment by eokwukwe** [comment]
The admin page is not opening in the browser. I keep getting `mailpit refused to connect.`

**Reply by rizaldywirawan** [reply]
Have you try to run it? By right click your Laragon, and choose the Laragon menu, and run it?

**Comment by Pen-y-Fan** [comment]
See the troubleshooting section [How to stop Mailpit](https://pen-y-fan.github.io/2023/02/23/how-to-install-mailpit-in-laragon/#how-to-stop-mailpit), run the command or check Task manager to see if mailpit is running.

If it isn't running navigate to mailpit using cmdr and start it manually. Check the output for any error messages: (adjust c: if you have installed it on a different drive)

```sh
cd c:\laragon\bin\mailpit
mailpit.exe
```

Check the output, something like this means Mailpit is already running or another program is using port 1025:

```text
INFO[2023/09/08 08:16:11] [smtp] starting on 0.0.0.0:1025
INFO[2023/09/08 08:16:11] [http] starting server on http://0.0.0.0:8025/
ERRO[2023/09/08 08:16:11] listen tcp 0.0.0.0:1025: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.
```

Let us know what error messages you are receiving.

**Reply by rescenic** [reply]
Thank you so much for the guide.

**Reply by rescenic** [reply]
Mailpit doesn't work for Laragon 4. If I update cmder, maybe something will be not working.

**Reply by rescenic** [reply]
I am upgrading Laragon from version 4 to version 6, only using laragon.exe binary (overwrite laragon.exe - copy the older for backup), and adapt the procfile.bat and add cmder in laragon/etc/cmder from version 6 to version 4. And then update cmder in bin folder. Now, Mailpit is working.

**Comment by dickygiancini** [comment]
@Pen-y-Fan I'm currently running my old company code that still uses PHPMailer (not Laravel), and when i try to follow your guide step by step, the smtp response would always be : `Mailer Error: SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting` , when i run mailpit as usual in my CMD it works fine though, is there a problem when integrating with Laragon? This is my current env:

MAIL_HOST=localhost
MAIL_PORT=1025
MAIL_USERNAME=""
MAIL_PASSWORD=""
MAIL_TLS_ENCRYPTION=false
MAIL_SSL_ENCRYPTION=false

I changed the MAIL_HOST to mailpit per your guide but the results are the same

**Reply by Pen-y-Fan** [reply]
See if you can run PHPMailer from a PHP script, see:

- https://gist.github.com/rogeriopradoj/c19f017c05bde0008185b0326125a114
- https://stackoverflow.com/questions/61220173/phpmailer-mailhog-connection-refused-host-not-known

**Reply by agilare** [reply]
I was in the same situation and I could make it work with

```php
$mailer = new PHPMailer();
// ...
$mailer->SMTPAuth = false;
```
